log.go

```diff
@@ -2,12 +2,119 @@ package didplc
 
 import (
 	"fmt"
+	"sync"
 	"time"
 
-	"github.com/bluesky-social/indigo/atproto/crypto"
 	"github.com/bluesky-social/indigo/atproto/syntax"
 )
 
+type OpStatus struct {
+	CreatedAt   time.Time // the only immutable field here
+	Nullified   bool
+	LastChild   string // CID
+	AllowedKeys []string
+}
+
+// Note: LogValidationContext is designed such that it could later be turned into an interface,
+// optionally backed by a db rather than in-memory
+// Note: ops are globally unique by CID, so opStatus map can be shared across all DIDs
+type LogValidationContext struct {
+	head     map[string]string    // DID -> CID, tracks most recent valid op for a particular DID
+	opStatus map[string]*OpStatus // CID -> OpStatus
```

Comment:
having these `string` be the syntax string wrapper types (`syntax.DID`, `syntax.CID`) would make this a bit more self-documenting and could prevent mistakes down the line. I think because they are just string wrapper types they would compile away w/o performance impact.

OTOH not sure the effort is needed

Reply:
I thought about this. I was worried about introducing overhead from calling `ParseCID()` everywhere (if we wrapped strings directly then we'd be adding new potential for future mistakes imho) - but the overhead is probably negligible in the grand scheme of things (will be dwarfed by hashing and sig verification)

Reply:
(`syntax.CID` doc says "Always use [ParseCID] instead of wrapping strings directly, especially when working with network input.")

Reply:
The docs do indeed say that, and calling `ParseCID()` does have overhead, but we can do the explicit wrapping if we are confident, especially in internal logic.

I don't think it is a blocking issue, and could be justified either way.